### ISSUE TITLE
Blocking / Timeout when returning >238 chars

### ISSUE THREAD

**Issue by zedage** [sijms/go-ora#408]
@sijms Based on your example in `udt_array/main.go` (which by the way works fine), I made more tests and I struggled when trying to return rows with some more data. After digging deeper, I could make a test set (see attachment), which works when returning <= 238 chars, but fails when returning >238 chars. Thanks for your investigations.

[main.go.gz](https://github.com/sijms/go-ora/files/12109013/main.go.gz)

```go
package main

import (
	"database/sql"
	"fmt"
	go_ora "github.com/sijms/go-ora/v2"
	"log"
)

type HkGoraIdObj struct {
	Id int64 `udt:"ID"`
}

type HkGoraTestObj struct {
	Id   int64  `udt:"ID"`
	Name string `udt:"NAME"`
}

func execSimpleStmt(con *sql.DB, stmts ...string) {
	for _, stmt := range stmts {
		if _, err := con.Exec(stmt); err != nil {
			log.Println("Error when executing statement", stmt, err)
		}
	}
}

func createTypes(con *sql.DB) func() {
	log.Println("creating types")
	execSimpleStmt(con,
		"create or replace type HK_GORA_ID_OBJ force as object(ID number(22,0));",
		"create or replace type HK_GORA_ID_COLL as table of HK_GORA_ID_OBJ;",
		"create or replace type HK_GORA_TEST_OBJ force as object(ID number(22,0), NAME varchar2(2000));",
		"create or replace type HK_GORA_TEST_COLL as table of HK_GORA_TEST_OBJ;")
	return func() {
		log.Println("dropping types")
		execSimpleStmt(con,
			"drop type HK_GORA_ID_COLL",
			"drop type HK_GORA_ID_OBJ",
			"drop type HK_GORA_TEST_COLL",
			"drop type HK_GORA_TEST_OBJ")
	}
}

func createPackage(con *sql.DB) func() {
	log.Println("creating package")
	execSimpleStmt(con, `
create or replace package HK_GORA_TEST_PCK as
   function F_GetData(P_Ids HK_GORA_ID_COLL, P_AnzChars number) return HK_GORA_TEST_COLL;
end;`, `
create or replace package body HK_GORA_TEST_PCK as
   function F_GetData(P_Ids HK_GORA_ID_COLL, P_AnzChars number) return HK_GORA_TEST_COLL is
      L_Coll HK_GORA_TEST_COLL := HK_GORA_TEST_COLL();
   begin
	  
      for r in (with data as(
                  select 1 as ID, rpad('a',P_AnzChars,'a') as DATA from dual union all
                  select 2      , 'Data2'                   from dual
               )
               select HK_GORA_TEST_OBJ(ID, DATA) as HK_GORA_DATA
                 from data 
                where ID in (select ID from table(P_Ids))) loop
         L_Coll.Extend;
         L_Coll(L_Coll.LAST) := r.HK_GORA_DATA;
      end loop;
      return L_Coll;
   end;
end;`)
	return func() {
		log.Println("dropping package")
		execSimpleStmt(con, `drop package HK_GORA_TEST_PCK`)
	}
}

func getConnection() (*sql.DB, func()) {
	connectString := go_ora.BuildUrl("hacki-aws.devops.f4o.ch", 1521, "nickel", "f2ldatp0", "fgth", nil)
	log.Println("connectString", fmt.Sprintf("Oracle connect string: %s", connectString))
	dbCon, err := sql.Open("oracle", connectString)
	if err != nil {
		log.Fatal(err)
	}
	return dbCon, func() {
		if err = dbCon.Close(); err != nil {
			log.Fatal(err)
		}
	}
}

func registerTypes(con *sql.DB) {
	err := go_ora.RegisterType(con, "HK_GORA_ID_OBJ", "HK_GORA_ID_COLL", HkGoraIdObj{})
	if err != nil {
		log.Fatal(err)
	}
	err = go_ora.RegisterType(con, "HK_GORA_TEST_OBJ", "HK_GORA_TEST_COLL", HkGoraTestObj{})
	if err != nil {
		log.Fatal(err)
	}
}
func executeTest(con *sql.DB, anzChars int) {
	if anzChars > 238 {
		log.Println("Executing a test that will block / timeout (>238 chars)")
	} else {
		log.Println("Execution a test without block or timeout (<=238 chars)")
	}

	var res []HkGoraTestObj
	con.Exec(`
begin
   :1 := HK_GORA_TEST_PCK.F_GetData(:2, :3);
end;`,
		go_ora.Out{Dest: &res, Size: 2},
		[]HkGoraIdObj{{Id: 1}, {Id: 2}},
		anzChars)
}

func main() {
	log.Println("Start test")
	con, deferFn := getConnection()
	defer deferFn()
	deferFn = createTypes(con)
	defer deferFn()
	deferFn = createPackage(con)
	defer deferFn()
	registerTypes(con)
	executeTest(con, 238)
	executeTest(con, 239)
}
```

**Comment by sijms** [comment]
I will investigate

**Comment by sijms** [comment]
fixed in next release 2.7.10
